Chore(docs): update Vale terms and contributing guide (#911)
midnightntwrk/midnight-docs: CONTRIBUTING.md

* **Understand the project**: Familiarize yourself with Midnight's architecture, technology, and coding standards. You can find relevant information in our litepaper.
* **Set up your development environment**: Ensure you have the necessary tools and dependencies installed. See our developer [documentation](https://docs.midnight.network/) for detailed instructions.

## Submit issues

Use one of the [issue templates](https://github.com/midnightntwrk/midnight-docs/issues/new/choose) to submit an issue to the Midnight docs repository. The Midnight team or a community member will address it if it's relevant.

Ensure the title is a clear summary of the requirement and provides enough context.

Supported issue types include:

- **Contribute a blog**: Suggest or submit a blog post for the Midnight community site.
- **Fix content**: Report issues with the Midnight documentation content.
- **New content**: Suggest new documentation content or request documentation for a new Midnight feature.
- **Site**: Request updates to the Midnight documentation site that affect its interface, functionality, or infrastructure.

> ## Documentation contribution process

Reviewer: This section doesn't actually describe a process.

Author: Could we say guidelines instead? 

The individual docs linked to from each list item describe the actual process. I didn't want to duplicate it here or make this doc super long.

Reviewer: Yes.

Before contributing to the doc, review the following guides:

- **[Doc contribution guide](https://github.com/midnightntwrk/midnight-docs/blob/main/docs/_contribute/contribute-docs.mdx)**: This includes a step-by-step guide on setting up your GitHub environment, creating pull requests, file naming convention.
- **[Style guide](https://github.com/midnightntwrk/midnight-docs/blob/main/docs/_contribute/style-guide.mdx)**: The style guide helps contributors adhere to writing and formatting standards for the Midnight blockchain.
- **[Formatting guide](https://github.com/midnightntwrk/midnight-docs/blob/main/docs/_contribute/formatting-guide.mdx)**: This includes a detailed guide on some of the markdown syntax and component used in the Midnight documentation.

> ## Code contribution process

Reviewer: This is not a process.

# Contribute

We welcome your contributions to the Midnight network! By contributing, you'll play a vital role in shaping the future of a blockchain focused on data privacy.

## Get started

> * **Review existing contributions and issues**: Before submitting, please check if a similar issue or feature request already exists by searching our issue tracker.

Reviewer 2: **Style (tone, voice)**: "please" reads as slightly patronizing (rule 7), and "our issue tracker" uses first person (rule 11). Since you're already touching this line:

```suggestion
* **Review existing contributions and issues**: Before submitting, check if a similar issue or feature request already exists by searching the issue tracker.
```

Author: @oduameh Thanks for the review and suggestions.

* **Understand the project**: Familiarize yourself with Midnight's architecture, technology, and coding standards. You can find relevant information in our litepaper.
* **Set up your development environment**: Ensure you have the necessary tools and dependencies installed. See our developer [documentation](https://docs.midnight.network/) for detailed instructions.

## Submit issues

> Use one of the [issue templates](https://github.com/midnightntwrk/midnight-docs/issues/new/choose) to submit an issue to the Midnight docs repository. The Midnight team or a community member will address it if it's relevant.

Reviewer 2: **Style (tense)**: "will address" uses future tense — the style guide prefers present tense for describing how things work (rule 9).

```suggestion
Use one of the [issue templates](https://github.com/midnightntwrk/midnight-docs/issues/new/choose) to submit an issue to the Midnight docs repository. The Midnight team or a community member triages and addresses relevant issues.
```

Ensure the title is a clear summary of the requirement and provides enough context.

Supported issue types include:

- **Contribute a blog**: Suggest or submit a blog post for the Midnight community site.
- **Fix content**: Report issues with the Midnight documentation content.
- **New content**: Suggest new documentation content or request documentation for a new Midnight feature.
- **Site**: Request updates to the Midnight documentation site that affect its interface, functionality, or infrastructure.

## Documentation contribution guidelines

Before contributing to the doc, review the following guides:

> - **[Doc contribution guide](https://github.com/midnightntwrk/midnight-docs/blob/main/docs/_contribute/contribute-docs.mdx)**: This includes a step-by-step guide on setting up your GitHub environment, creating pull requests, file naming convention, and more.

Reviewer 2: **Style (precision)**: "file naming convention" should be plural ("conventions"), and "and more" is vague — the style guide values specificity (rule 1).

```suggestion
- **[Doc contribution guide](https://github.com/midnightntwrk/midnight-docs/blob/main/docs/_contribute/contribute-docs.mdx)**: A step-by-step guide to setting up your GitHub environment, creating pull requests, and following file naming conventions.
```

- **[Style guide](https://github.com/midnightntwrk/midnight-docs/blob/main/docs/_contribute/style-guide.mdx)**: The style guide helps contributors adhere to writing and formatting standards for the Midnight blockchain.
- **[Formatting guide](https://github.com/midnightntwrk/midnight-docs/blob/main/docs/_contribute/formatting-guide.mdx)**: This includes a detailed guide on markdown syntax and components used in the Midnight documentation.

## Code contribution guidelines

All code contributions must follow these guidelines. Adhering to them helps maintainers review your changes efficiently and keeps the review history clean.

> * **Pull requests**: Code contributions are submitted via pull requests.

Reviewer 2: **Style (voice)**: "Code contributions are submitted" is passive voice (rule 8). Flip to active:

```suggestion
* **Pull requests**: Submit code contributions via pull requests.
```

* **Fork the repository**: Create your own fork of the Midnight docs repository.
* **Create a branch**: Make your changes in a separate branch, prefixed with a short name moniker (for example, `jill-my-feature`).
* **Follow coding standards**: Adhere to the coding style guides specified in our documentation.
* **Write tests**: Include unit tests and integration tests to cover your changes.
* **Commit messages**: Write clear and concise commit messages.
* **Submit pull request**: Submit your pull request to the appropriate branch in the main repository.
* **Do not `--force` push**: Doing so requires reviewers to re-review all commits in the PR rather than only the commits since the last review.
* **Code review**: All pull requests undergo code review by project maintainers. Be prepared to address feedback from reviewers.

## Requirements for acceptable contributions

> To ensure that your contributions are reviewed and merged, you need to adhere to the following standards:

Reviewer 2: **Style (voice, concision)**: "are reviewed and merged" is passive, and "you need to adhere to the following standards" is wordy. Simplify:

```suggestion
Follow these standards so your contributions can be reviewed and merged:
```

> * **Coding**: Code must adhere to the coding style guides defined in our documentation

Reviewer 2: **Style (voice, formatting)**: "our documentation" uses first person (rule 11), and the line is missing a period at the end.

```suggestion
* **Coding**: Code must adhere to the coding style guides defined in the project documentation.
```

* **Code Review:** All pull requests undergo code review by project maintainers.
  Be prepared to address feedback from reviewers.

## Requirements for acceptable contributions

> To ensure your contributions are reviewed and merged, you need to adhere to the following:

Reviewer: ```suggestion
To ensure that your contributions are reviewed and merged, you must adhere to the following:
```

Reviewer: The following what? Rules? Standards?

Author: Good catch. Updated
